framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -1934,23 +1934,23 @@ private boolean isSyntheticArgument(Tree tree) {
      */
     protected final AnnotatedDeclaredType fromNewClass(NewClassTree newClassTree) {
         AnnotatedDeclaredType type;
-        if (!TreeUtils.isDiamondTree(newClassTree)) {
-            // If newClassTree does not create anonymous class,
-            // newClassTree.getIdentifier includes the explicit annotations in this location:
-            //   new @HERE Class()
-            type = (AnnotatedDeclaredType) fromTypeTree(newClassTree.getIdentifier());
-        } else {
-            type = (AnnotatedDeclaredType) toAnnotatedType(InternalUtils.typeOf(newClassTree), false);
-        }
 
         if (newClassTree.getClassBody() != null) {
+            type = (AnnotatedDeclaredType) toAnnotatedType(InternalUtils.typeOf(newClassTree), false);
```

Comment:
Can an anonymous class use diamonds?
I don't quite understand the three cases and why diamonds are handled again on line 1956.